presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -520,15 +520,13 @@ private static boolean isDirectory(HdfsContext context, HdfsEnvironment hdfsEnvi
     public static Path createTemporaryPath(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path targetPath)
     {
         // use a per-user temporary directory to avoid permission problems
-        String temporaryPrefix = "/tmp/presto-" + context.getIdentity().getUser();
+        Path temporaryRoot = new Path("/tmp/presto-" + context.getIdentity().getUser());
 
         // use relative temporary directory on ViewFS
         if (isViewFileSystem(context, hdfsEnvironment, targetPath)) {
-            temporaryPrefix = ".hive-staging";
+            temporaryRoot = new Path(targetPath.getParent(), ".hive-staging");
         }
 
-        // create a temporary directory on the same filesystem
-        Path temporaryRoot = new Path(targetPath, temporaryPrefix);
```

Comment:
I think this breaks the non-viewfs case: it will set the temporary root to `/tmp` rather than the target file system.

Reply:
thx, @electrum , yes, it's confusing for non-viewfs case, I had done a test 
    Path a = new Path("/a");
    Path b = new Path("/b");
    LOG.info((new Path(a, b).toString()));
it print "/b" not "/a/b", 
so for the non-viewfs case, the root is /tmp, yes?

Reply:
@electrum / @sandflee - maybe an option is to just have this line as your fix?
```
Path temporaryRoot = new Path(targetPath.getParent(), temporaryPrefix);
```
That way in both the viewfs case and non-viewfs case, we create the directory temporaryPrefix in the parent of the targetPath so we don't end up creating the targetPath directory on HDFS and don't run into these issues.

Reply:
I just tried this change - https://github.com/piyushnarang/presto/commit/53052716ace86c2e9a94095ee7d9f13a404dd0d0 on our setup and I'm able to run the create table & ctas successfully now.